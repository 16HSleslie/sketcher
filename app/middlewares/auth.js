const jwt = require('jsonwebtoken');
const config = require('../../config');

module.exports = function(req, res, next) {
  // Get token from header
  const token = req.header('x-auth-token');

  // Check if no token
  if (!token) {
    return res.status(401).json({ msg: 'No token, authorization denied' });
  }

  try {
    // Verify token
    const decoded = jwt.verify(token, config.JWT_SECRET);

    // Add admin and user from payload to request
    // This handles both admin and user tokens for broader compatibility
    if (decoded.admin) {
      req.admin = decoded.admin;
      // Also set user for compatibility with code that expects req.user
      req.user = { id: decoded.admin.id };
    } else if (decoded.user) {
      req.user = decoded.user;
      // Also set admin for compatibility with code that expects req.admin
      req.admin = { id: decoded.user.id };
    }
    
    next();
  } catch (err) {
    res.status(401).json({ msg: 'Token is not valid' });
  }
}; 