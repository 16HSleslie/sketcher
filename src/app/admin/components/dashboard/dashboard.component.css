.admin-container {
  display: flex;
  min-height: 100vh;
  background-color: #f8f5f0;
}

/* Sidebar Styling */
.admin-sidebar {
  width: 250px;
  background-color: #5c4a38;
  color: #fff;
  display: flex;
  flex-direction: column;
  box-shadow: 3px 0 10px rgba(0, 0, 0, 0.1);
}

.sidebar-header {
  padding: 1.5rem;
  text-align: center;
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
}

.sidebar-header h1 {
  font-family: 'Cinzel', serif;
  font-size: 1.5rem;
  margin-bottom: 0.5rem;
  color: #f3e9d7;
}

.sidebar-header h2 {
  font-size: 0.9rem;
  font-weight: 400;
  color: #d0c8b5;
  letter-spacing: 1px;
}

.sidebar-nav {
  padding: 1.5rem 0;
  flex-grow: 1;
}

.nav-item {
  display: flex;
  align-items: center;
  padding: 0.75rem 1.5rem;
  color: #d0c8b5;
  text-decoration: none;
  transition: all 0.3s;
  position: relative;
}

.nav-item:hover, .nav-item.active {
  background-color: rgba(255, 255, 255, 0.1);
  color: #fff;
}

.nav-item.active::before {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  width: 4px;
  background-color: #a38868;
}

.nav-icon {
  width: 20px;
  height: 20px;
  margin-right: 12px;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  opacity: 0.8;
}

.nav-item:hover .nav-icon, .nav-item.active .nav-icon {
  opacity: 1;
}

.dashboard-icon {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23d0c8b5' viewBox='0 0 24 24'%3E%3Cpath d='M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z'/%3E%3C/svg%3E");
}

.products-icon {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23d0c8b5' viewBox='0 0 24 24'%3E%3Cpath d='M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 18H6V4h5v8l2.5-1.5L16 12V4h2v16z'/%3E%3C/svg%3E");
}

.orders-icon {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23d0c8b5' viewBox='0 0 24 24'%3E%3Cpath d='M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z'/%3E%3C/svg%3E");
}

.sidebar-footer {
  padding: 1rem 1.5rem;
  border-top: 1px solid rgba(255, 255, 255, 0.1);
}

.logout-button {
  display: flex;
  align-items: center;
  background: none;
  border: none;
  color: #d0c8b5;
  width: 100%;
  padding: 0.5rem;
  cursor: pointer;
  transition: all 0.3s;
  font-size: 0.9rem;
}

.logout-button:hover {
  color: #fff;
}

.logout-icon {
  width: 18px;
  height: 18px;
  margin-right: 10px;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23d0c8b5' viewBox='0 0 24 24'%3E%3Cpath d='M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z'/%3E%3C/svg%3E");
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  opacity: 0.8;
}

.logout-button:hover .logout-icon {
  opacity: 1;
}

/* Main Content Area */
.admin-content {
  flex-grow: 1;
  padding: 2rem;
  overflow-y: auto;
}

.content-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 2rem;
}

.content-header h1 {
  font-family: 'Cinzel', serif;
  color: #5c4a38;
  font-size: 1.8rem;
  margin: 0;
}

.user-info {
  color: #8b6c4f;
  font-size: 0.95rem;
}

/* Dashboard Stats */
.stats-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(240px, 1fr));
  gap: 1.5rem;
  margin-bottom: 2rem;
}

.stats-card {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.05);
  padding: 1.5rem;
  display: flex;
  align-items: center;
  border-left: 4px solid #8b6c4f;
}

.stats-icon {
  width: 48px;
  height: 48px;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  margin-right: 1rem;
}

.products-icon {
  background-color: #f0e6d7;
  border-radius: 50%;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%238b6c4f' viewBox='0 0 24 24'%3E%3Cpath d='M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 18H6V4h5v8l2.5-1.5L16 12V4h2v16z'/%3E%3C/svg%3E");
}

.orders-icon {
  background-color: #f0e6d7;
  border-radius: 50%;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%238b6c4f' viewBox='0 0 24 24'%3E%3Cpath d='M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z'/%3E%3C/svg%3E");
}

.pending-icon {
  background-color: #f0e6d7;
  border-radius: 50%;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%238b6c4f' viewBox='0 0 24 24'%3E%3Cpath d='M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z'/%3E%3C/svg%3E");
}

.stats-info h3 {
  color: #8b6c4f;
  margin: 0 0 0.25rem;
  font-size: 0.9rem;
  font-weight: 600;
}

.stats-value {
  color: #5c4a38;
  font-size: 1.8rem;
  font-weight: 700;
  margin: 0;
}

/* Dashboard Sections */
.dashboard-section {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.05);
  margin-bottom: 2rem;
}

.section-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1.25rem 1.5rem;
  border-bottom: 1px solid #f0e6d7;
}

.section-header h2 {
  color: #5c4a38;
  margin: 0;
  font-size: 1.2rem;
  font-weight: 600;
}

.view-all {
  color: #8b6c4f;
  text-decoration: none;
  font-size: 0.9rem;
  transition: color 0.3s;
}

.view-all:hover {
  color: #5c4a38;
  text-decoration: underline;
}

/* Tables */
.table-container {
  padding: 1rem 1.5rem;
}

.data-table {
  width: 100%;
  border-collapse: collapse;
}

.data-table th {
  text-align: left;
  padding: 0.75rem 1rem;
  color: #8b6c4f;
  font-weight: 600;
  font-size: 0.9rem;
  border-bottom: 2px solid #f0e6d7;
}

.data-table td {
  padding: 0.75rem 1rem;
  border-bottom: 1px solid #f0e6d7;
  color: #5c4a38;
  font-size: 0.95rem;
}

.data-table tbody tr:hover {
  background-color: #fcfaf5;
}

.status-badge {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  border-radius: 4px;
  font-size: 0.8rem;
  font-weight: 500;
  text-transform: capitalize;
}

.status-badge.pending {
  background-color: #fff8e6;
  color: #d4a72c;
}

.status-badge.processing {
  background-color: #e6f7ff;
  color: #2c8fd4;
}

.status-badge.shipped {
  background-color: #e6ffe6;
  color: #2ca12c;
}

.status-badge.delivered {
  background-color: #e6ffe6;
  color: #2ca12c;
}

.status-badge.cancelled {
  background-color: #ffe6e6;
  color: #d42c2c;
}

.stock-badge {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  border-radius: 4px;
  font-size: 0.8rem;
  font-weight: 500;
  background-color: #fff8e6;
  color: #d4a72c;
}

.stock-badge.critical {
  background-color: #ffe6e6;
  color: #d42c2c;
}

/* Empty States */
.empty-state {
  padding: 2rem;
  text-align: center;
  color: #8b6c4f;
}

/* Loading State */
.loading-state {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 3rem;
  color: #8b6c4f;
}

.spinner {
  width: 40px;
  height: 40px;
  border: 3px solid #f0e6d7;
  border-top: 3px solid #8b6c4f;
  border-radius: 50%;
  margin-bottom: 1rem;
  animation: spin 1s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
} 