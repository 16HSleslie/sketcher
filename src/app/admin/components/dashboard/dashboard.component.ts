import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { AdminAuthService } from '../../services/auth.service';
import { ProductService } from '../../services/product.service';
import { OrderService, Order } from '../../services/order.service';

interface AdminStats {
  totalProducts: number;
  totalOrders: number;
  pendingOrders: number;
  recentOrders: Order[];
  lowStockProducts: any[];
}

@Component({
  selector: 'app-admin-dashboard',
  templateUrl: './dashboard.component.html',
  styleUrls: ['./dashboard.component.css']
})
export class AdminDashboardComponent implements OnInit {
  stats: AdminStats = {
    totalProducts: 0,
    totalOrders: 0,
    pendingOrders: 0,
    recentOrders: [],
    lowStockProducts: []
  };
  
  isLoading: boolean = true;
  adminProfile: any = null;

  constructor(
    private authService: AdminAuthService,
    private productService: ProductService,
    private orderService: OrderService,
    private router: Router
  ) {}

  ngOnInit(): void {
    this.loadAdminProfile();
    this.loadDashboardData();
  }

  loadAdminProfile(): void {
    this.authService.getAdminProfile().subscribe({
      next: (profile) => {
        this.adminProfile = profile;
      },
      error: (error) => {
        console.error('Error loading admin profile:', error);
        if (error.status === 401) {
          this.authService.logout();
          this.router.navigate(['/admin/login']);
        }
      }
    });
  }

  loadDashboardData(): void {
    this.isLoading = true;
    
    // Load products
    this.productService.getProducts().subscribe({
      next: (products) => {
        this.stats.totalProducts = products.length;
        this.stats.lowStockProducts = products
          .filter(product => product.stock < 5 && product.isAvailable)
          .slice(0, 5);
      },
      error: (error) => console.error('Error loading products:', error)
    });
    
    // Load orders
    this.orderService.getOrders().subscribe({
      next: (orders) => {
        this.stats.totalOrders = orders.length;
        this.stats.pendingOrders = orders.filter(order => 
          order.status === 'pending' || order.status === 'processing'
        ).length;
        
        // Get 5 most recent orders
        this.stats.recentOrders = orders
          .sort((a, b) => new Date(b.createdAt || 0).getTime() - new Date(a.createdAt || 0).getTime())
          .slice(0, 5);
          
        this.isLoading = false;
      },
      error: (error) => {
        console.error('Error loading orders:', error);
        this.isLoading = false;
      }
    });
  }

  logout(): void {
    this.authService.logout();
    this.router.navigate(['/admin/login']);
  }
} 