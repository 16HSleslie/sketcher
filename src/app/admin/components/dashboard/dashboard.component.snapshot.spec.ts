import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Router } from '@angular/router';
import { of } from 'rxjs';
import { AdminDashboardComponent } from './dashboard.component';
import { AdminAuthService } from '../../services/auth.service';
import { ProductService } from '../../services/product.service';
import { OrderService } from '../../services/order.service';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('AdminDashboardComponent Snapshots', () => {
  let component: AdminDashboardComponent;
  let fixture: ComponentFixture<AdminDashboardComponent>;
  let authServiceMock: jasmine.SpyObj<AdminAuthService>;
  let productServiceMock: jasmine.SpyObj<ProductService>;
  let orderServiceMock: jasmine.SpyObj<OrderService>;
  let routerMock: jasmine.SpyObj<Router>;

  beforeEach(async () => {
    // Create spy objects for services
    authServiceMock = jasmine.createSpyObj('AdminAuthService', ['getAdminProfile', 'logout']);
    productServiceMock = jasmine.createSpyObj('ProductService', ['getProducts']);
    orderServiceMock = jasmine.createSpyObj('OrderService', ['getOrders']);
    routerMock = jasmine.createSpyObj('Router', ['navigate']);

    // Configure default behavior for mocks
    authServiceMock.getAdminProfile.and.returnValue(of({
      _id: '1',
      username: 'admin',
      email: 'admin@example.com',
      lastLogin: new Date('2023-01-01').toISOString() // Fixed date for stable snapshots
    }));
    
    productServiceMock.getProducts.and.returnValue(of([
      {
        _id: '1',
        name: 'Product 1',
        description: 'Description 1',
        price: 100,
        images: ['img1.jpg'],
        category: 'Books',
        stock: 10,
        isAvailable: true
      },
      {
        _id: '2',
        name: 'Product 2',
        description: 'Description 2',
        price: 150,
        images: ['img2.jpg'],
        category: 'Books',
        stock: 3,
        isAvailable: true
      }
    ]));
    
    orderServiceMock.getOrders.and.returnValue(of([
      {
        _id: '1',
        customer: { name: 'Customer 1', email: 'customer1@example.com' },
        items: [{ product: '1', quantity: 1, price: 100 }],
        isCustomOrder: false,
        total: 100,
        status: 'completed',
        paymentMethod: 'credit_card',
        paymentStatus: 'completed',
        createdAt: new Date('2023-01-01').toISOString() // Fixed date for stable snapshots
      },
      {
        _id: '2',
        customer: { name: 'Customer 2', email: 'customer2@example.com' },
        items: [{ product: '2', quantity: 1, price: 150 }],
        isCustomOrder: false,
        total: 150,
        status: 'pending',
        paymentMethod: 'paypal',
        paymentStatus: 'pending',
        createdAt: new Date('2023-01-02').toISOString() // Fixed date for stable snapshots
      }
    ]));

    await TestBed.configureTestingModule({
      declarations: [ AdminDashboardComponent ],
      providers: [
        { provide: AdminAuthService, useValue: authServiceMock },
        { provide: ProductService, useValue: productServiceMock },
        { provide: OrderService, useValue: orderServiceMock },
        { provide: Router, useValue: routerMock }
      ],
      schemas: [NO_ERRORS_SCHEMA] // Ignore unknown elements for snapshot testing
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AdminDashboardComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match snapshot for dashboard component', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot for dashboard with low stock warning', () => {
    // Set a product to low stock to trigger warning display
    component.stats.lowStockProducts = [
      {
        _id: '2',
        name: 'Product 2',
        description: 'Description 2',
        price: 150,
        images: ['img2.jpg'],
        category: 'Books',
        stock: 1, // Very low stock
        isAvailable: true
      }
    ];
    
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot for dashboard with pending orders', () => {
    // Set pending orders
    component.stats.pendingOrders = 5;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot for loading state', () => {
    component.isLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
}); 