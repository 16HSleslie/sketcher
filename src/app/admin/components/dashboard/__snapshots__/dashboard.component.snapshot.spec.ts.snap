// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboardComponent Snapshots should match snapshot for dashboard component 1`] = `
<app-admin-dashboard
  adminProfile={[Function Object]}
  authService={[Function Object]}
  isLoading="false"
  orderService={[Function Object]}
  productService={[Function Object]}
  router={[Function Object]}
  stats={[Function Object]}
>
  <div
    class="admin-container"
  >
    <div
      class="admin-sidebar"
    >
      <div
        class="sidebar-header"
      >
        <h1>
          Lily's Bookbinding
        </h1>
        <h2>
          Admin Portal
        </h2>
      </div>
      <nav
        class="sidebar-nav"
      >
        <a
          class="nav-item"
          routerlink="/admin/dashboard"
          routerlinkactive="active"
        >
          <i
            class="nav-icon dashboard-icon"
          />
          <span>
            Dashboard
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/products"
          routerlinkactive="active"
        >
          <i
            class="nav-icon products-icon"
          />
          <span>
            Products
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/orders"
          routerlinkactive="active"
        >
          <i
            class="nav-icon orders-icon"
          />
          <span>
            Orders
          </span>
        </a>
      </nav>
      <div
        class="sidebar-footer"
      >
        <button
          class="logout-button"
        >
          <i
            class="logout-icon"
          />
          <span>
            Log Out
          </span>
        </button>
      </div>
    </div>
    <div
      class="admin-content"
    >
      <div
        class="content-header"
      >
        <h1>
          Dashboard
        </h1>
        <div
          class="user-info"
        >
          <span>
            Welcome, admin
          </span>
        </div>
      </div>
      <div
        class="dashboard-content"
      >
        <div
          class="stats-grid"
        >
          <div
            class="stats-card"
          >
            <div
              class="stats-icon products-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Products
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon orders-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Orders
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon pending-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Pending Orders
              </h3>
              <p
                class="stats-value"
              >
                1
              </p>
            </div>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Recent Orders
            </h2>
            <a
              class="view-all"
              routerlink="/admin/orders"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Order ID
                  </th>
                  <th>
                    Customer
                  </th>
                  <th>
                    Date
                  </th>
                  <th>
                    Total
                  </th>
                  <th>
                    Status
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    2...
                  </td>
                  <td>
                    Customer 2
                  </td>
                  <td>
                    Jan 2, 2023
                  </td>
                  <td>
                    $150.00
                  </td>
                  <td>
                    <span
                      class="status-badge pending"
                    >
                       pending 
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    1...
                  </td>
                  <td>
                    Customer 1
                  </td>
                  <td>
                    Jan 1, 2023
                  </td>
                  <td>
                    $100.00
                  </td>
                  <td>
                    <span
                      class="status-badge completed"
                    >
                       completed 
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Low Stock Products
            </h2>
            <a
              class="view-all"
              routerlink="/admin/products"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Product
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Stock
                  </th>
                  <th>
                    Price
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Product 2
                  </td>
                  <td>
                    Books
                  </td>
                  <td>
                    <span
                      class="stock-badge"
                    >
                       3 
                    </span>
                  </td>
                  <td>
                    $150.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</app-admin-dashboard>
`;

exports[`AdminDashboardComponent Snapshots should match snapshot for dashboard with low stock warning 1`] = `
<app-admin-dashboard
  adminProfile={[Function Object]}
  authService={[Function Object]}
  isLoading="false"
  orderService={[Function Object]}
  productService={[Function Object]}
  router={[Function Object]}
  stats={[Function Object]}
>
  <div
    class="admin-container"
  >
    <div
      class="admin-sidebar"
    >
      <div
        class="sidebar-header"
      >
        <h1>
          Lily's Bookbinding
        </h1>
        <h2>
          Admin Portal
        </h2>
      </div>
      <nav
        class="sidebar-nav"
      >
        <a
          class="nav-item"
          routerlink="/admin/dashboard"
          routerlinkactive="active"
        >
          <i
            class="nav-icon dashboard-icon"
          />
          <span>
            Dashboard
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/products"
          routerlinkactive="active"
        >
          <i
            class="nav-icon products-icon"
          />
          <span>
            Products
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/orders"
          routerlinkactive="active"
        >
          <i
            class="nav-icon orders-icon"
          />
          <span>
            Orders
          </span>
        </a>
      </nav>
      <div
        class="sidebar-footer"
      >
        <button
          class="logout-button"
        >
          <i
            class="logout-icon"
          />
          <span>
            Log Out
          </span>
        </button>
      </div>
    </div>
    <div
      class="admin-content"
    >
      <div
        class="content-header"
      >
        <h1>
          Dashboard
        </h1>
        <div
          class="user-info"
        >
          <span>
            Welcome, admin
          </span>
        </div>
      </div>
      <div
        class="dashboard-content"
      >
        <div
          class="stats-grid"
        >
          <div
            class="stats-card"
          >
            <div
              class="stats-icon products-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Products
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon orders-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Orders
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon pending-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Pending Orders
              </h3>
              <p
                class="stats-value"
              >
                1
              </p>
            </div>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Recent Orders
            </h2>
            <a
              class="view-all"
              routerlink="/admin/orders"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Order ID
                  </th>
                  <th>
                    Customer
                  </th>
                  <th>
                    Date
                  </th>
                  <th>
                    Total
                  </th>
                  <th>
                    Status
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    2...
                  </td>
                  <td>
                    Customer 2
                  </td>
                  <td>
                    Jan 2, 2023
                  </td>
                  <td>
                    $150.00
                  </td>
                  <td>
                    <span
                      class="status-badge pending"
                    >
                       pending 
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    1...
                  </td>
                  <td>
                    Customer 1
                  </td>
                  <td>
                    Jan 1, 2023
                  </td>
                  <td>
                    $100.00
                  </td>
                  <td>
                    <span
                      class="status-badge completed"
                    >
                       completed 
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Low Stock Products
            </h2>
            <a
              class="view-all"
              routerlink="/admin/products"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Product
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Stock
                  </th>
                  <th>
                    Price
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Product 2
                  </td>
                  <td>
                    Books
                  </td>
                  <td>
                    <span
                      class="stock-badge critical"
                    >
                       1 
                    </span>
                  </td>
                  <td>
                    $150.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</app-admin-dashboard>
`;

exports[`AdminDashboardComponent Snapshots should match snapshot for dashboard with pending orders 1`] = `
<app-admin-dashboard
  adminProfile={[Function Object]}
  authService={[Function Object]}
  isLoading="false"
  orderService={[Function Object]}
  productService={[Function Object]}
  router={[Function Object]}
  stats={[Function Object]}
>
  <div
    class="admin-container"
  >
    <div
      class="admin-sidebar"
    >
      <div
        class="sidebar-header"
      >
        <h1>
          Lily's Bookbinding
        </h1>
        <h2>
          Admin Portal
        </h2>
      </div>
      <nav
        class="sidebar-nav"
      >
        <a
          class="nav-item"
          routerlink="/admin/dashboard"
          routerlinkactive="active"
        >
          <i
            class="nav-icon dashboard-icon"
          />
          <span>
            Dashboard
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/products"
          routerlinkactive="active"
        >
          <i
            class="nav-icon products-icon"
          />
          <span>
            Products
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/orders"
          routerlinkactive="active"
        >
          <i
            class="nav-icon orders-icon"
          />
          <span>
            Orders
          </span>
        </a>
      </nav>
      <div
        class="sidebar-footer"
      >
        <button
          class="logout-button"
        >
          <i
            class="logout-icon"
          />
          <span>
            Log Out
          </span>
        </button>
      </div>
    </div>
    <div
      class="admin-content"
    >
      <div
        class="content-header"
      >
        <h1>
          Dashboard
        </h1>
        <div
          class="user-info"
        >
          <span>
            Welcome, admin
          </span>
        </div>
      </div>
      <div
        class="dashboard-content"
      >
        <div
          class="stats-grid"
        >
          <div
            class="stats-card"
          >
            <div
              class="stats-icon products-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Products
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon orders-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Total Orders
              </h3>
              <p
                class="stats-value"
              >
                2
              </p>
            </div>
          </div>
          <div
            class="stats-card"
          >
            <div
              class="stats-icon pending-icon"
            />
            <div
              class="stats-info"
            >
              <h3>
                Pending Orders
              </h3>
              <p
                class="stats-value"
              >
                5
              </p>
            </div>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Recent Orders
            </h2>
            <a
              class="view-all"
              routerlink="/admin/orders"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Order ID
                  </th>
                  <th>
                    Customer
                  </th>
                  <th>
                    Date
                  </th>
                  <th>
                    Total
                  </th>
                  <th>
                    Status
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    2...
                  </td>
                  <td>
                    Customer 2
                  </td>
                  <td>
                    Jan 2, 2023
                  </td>
                  <td>
                    $150.00
                  </td>
                  <td>
                    <span
                      class="status-badge pending"
                    >
                       pending 
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    1...
                  </td>
                  <td>
                    Customer 1
                  </td>
                  <td>
                    Jan 1, 2023
                  </td>
                  <td>
                    $100.00
                  </td>
                  <td>
                    <span
                      class="status-badge completed"
                    >
                       completed 
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="dashboard-section"
        >
          <div
            class="section-header"
          >
            <h2>
              Low Stock Products
            </h2>
            <a
              class="view-all"
              routerlink="/admin/products"
            >
              View All
            </a>
          </div>
          <div
            class="table-container"
          >
            <table
              class="data-table"
            >
              <thead>
                <tr>
                  <th>
                    Product
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Stock
                  </th>
                  <th>
                    Price
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Product 2
                  </td>
                  <td>
                    Books
                  </td>
                  <td>
                    <span
                      class="stock-badge"
                    >
                       3 
                    </span>
                  </td>
                  <td>
                    $150.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</app-admin-dashboard>
`;

exports[`AdminDashboardComponent Snapshots should match snapshot for loading state 1`] = `
<app-admin-dashboard
  adminProfile={[Function Object]}
  authService={[Function Object]}
  isLoading={[Function Boolean]}
  orderService={[Function Object]}
  productService={[Function Object]}
  router={[Function Object]}
  stats={[Function Object]}
>
  <div
    class="admin-container"
  >
    <div
      class="admin-sidebar"
    >
      <div
        class="sidebar-header"
      >
        <h1>
          Lily's Bookbinding
        </h1>
        <h2>
          Admin Portal
        </h2>
      </div>
      <nav
        class="sidebar-nav"
      >
        <a
          class="nav-item"
          routerlink="/admin/dashboard"
          routerlinkactive="active"
        >
          <i
            class="nav-icon dashboard-icon"
          />
          <span>
            Dashboard
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/products"
          routerlinkactive="active"
        >
          <i
            class="nav-icon products-icon"
          />
          <span>
            Products
          </span>
        </a>
        <a
          class="nav-item"
          routerlink="/admin/orders"
          routerlinkactive="active"
        >
          <i
            class="nav-icon orders-icon"
          />
          <span>
            Orders
          </span>
        </a>
      </nav>
      <div
        class="sidebar-footer"
      >
        <button
          class="logout-button"
        >
          <i
            class="logout-icon"
          />
          <span>
            Log Out
          </span>
        </button>
      </div>
    </div>
    <div
      class="admin-content"
    >
      <div
        class="content-header"
      >
        <h1>
          Dashboard
        </h1>
        <div
          class="user-info"
        >
          <span>
            Welcome, admin
          </span>
        </div>
      </div>
      <div
        class="loading-state"
      >
        <div
          class="spinner"
        />
        <p>
          Loading dashboard data...
        </p>
      </div>
    </div>
  </div>
</app-admin-dashboard>
`;
